jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/internal/AnyUserLoginService.java

```diff
@@ -0,0 +1,115 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe.internal;
+
+import java.util.function.Function;
+import javax.security.auth.Subject;
+
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.UserIdentity;
+import org.eclipse.jetty.security.UserPrincipal;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Session;
+
+/**
+ * A {@link LoginService} which allows any user to be authenticated.
```

Comment:
```suggestion
 * A {@link LoginService} which allows unknown users to be authenticated without checking credentials.
 * It is assumed that the credentials are already checked by the Authenticator.
```

Reply:
I included the `unknown users` part but not about the `without checking credentials`.

Doesn't quite make sense in this context as there are never any credentials for SIWE.
And it would check credentials against the nested loginService for known users, but not for unknown users, so this would need to be clarified.